iina/Extensions.swift

```diff
@@ -316,3 +316,13 @@ extension Data {
     return fileUrl
   }
 }
+
+
+extension CharacterSet {
+  static let urlAllowed = CharacterSet.urlHostAllowed
```

Comment:
Yes that would be the best solution, but what user want initially is like "percent encoding the spaces" in URL. `NSURLComponent` won't recognize the url if it contains spaces, so if we don't add percent encoding at first, we must split it manually. Plus mpv is doing the same thing, so I think It's OK to stay consistent with mpv.